autoload/neomake/makers/ft/rust.vim

```diff
@@ -135,16 +135,26 @@ function! neomake#makers#ft#rust#CargoProcessOutput(context) abort
         endif
 
         let error = {'maker_name': 'cargo'}
-        let code_dict = get(data, 'code', -1)
-        if code_dict is g:neomake#compat#json_null
-            if get(data, 'level', '') ==# 'warning'
-                let error.type = 'W'
-            else
-                let error.type = 'E'
+        let target = get(decoded, 'target', -1)
+        if type(target) == type({})
+            let filename = get(target, 'src_path', -1)
+            if type(filename) == type('')
+                let error.filename = filename
             endif
```

Comment:
Cool.  I assume this fixes/closes https://github.com/neomake/neomake/issues/2085 then.

Reply:
yeah, as far as I understand